truffle/src/main/java/org/jruby/truffle/nodes/core/KernelNodes.java

```diff
@@ -945,14 +949,42 @@ public boolean isANil(RubyBasicObject self, Object nil) {
             return false;
         }
 
-        @TruffleBoundary
+        @Specialization(
+                limit = "getCacheLimit()",
+                guards = {"getMetaClass(frame, self) == cachedMetaClass", "module == cachedModule"},
+                assumptions = "cachedModule.getUnmodifiedAssumption()")
+        public boolean isACached(VirtualFrame frame,
+                                 Object self,
+                                 RubyModule module,
+                                 @Cached("getMetaClass(frame, self)") RubyClass cachedMetaClass,
+                                 @Cached("module") RubyModule cachedModule,
+                                 @Cached("isA(cachedMetaClass, cachedModule)") boolean result) {
+            return result;
+        }
+
         @Specialization
-        public boolean isA(Object self, RubyModule rubyClass) {
+        public boolean isAUncached(VirtualFrame frame, Object self, RubyModule module) {
```

Comment:
I guess I'm losing my mind.  I swear I read `isaUncached` and I checked a few times before commenting.  Sorry for the noice.